File: coding/lexer_module/go/try_lexer.go
package main

import (
	"fmt"
	"strconv"
	"./bklexer"
)

func main() {
	fmt.Println("Test Code:")
	code := "声明 变量 = PI * 100 - fda\n1024 * 4 * 3.14 ### \n123"
	fmt.Println(code)
	fmt.Println("--------------------------------")

	lexer := BKLexer.NewLexer()
	lexer.AddRule("\\d+\\.\\d*", "FLOAT")
	lexer.AddRule("\\d+", "INT")
	lexer.AddRule("[\\p{L}\\d_]+", "NAME")
	lexer.AddRule("\\+", "PLUS")
	lexer.AddRule("\\-", "MINUS")
	lexer.AddRule("\\*", "MUL")
	lexer.AddRule("/", "DIV")
	lexer.AddRule("=", "ASSIGN")
	lexer.AddRule("#[^\\r\\n]*", "COMMENT")
	lexer.AddIgnores("[ \\f\\t]+")

	lexer.Build(code)
	for true {
		token := lexer.NextToken()
		if (token.TType != BKLexer.TOKEN_TYPE_EOF) {
			fmt.Printf("%s\t%s\tt%d\t%d\t%d,%d\n",
			token.Name, strconv.Quote(token.Source), token.TType, token.Position, token.Row, token.Col)
		}
		if (token.TType == BKLexer.TOKEN_TYPE_EOF || token.TType == BKLexer.TOKEN_TYPE_ERROR) {
			break
		}
	}
}

File: coding/lexer_module/python/try_lexer.py
from bklexer import Token, Lexer


def main():
    print("Test Code:")
    code = "声明 变量 = PI * 100 - fda\n1024 * 4 * 3.14 ### \n123"
    print(code)
    print("--------------------------------")

    lexer = Lexer()
    lexer.add_rule("\\d+\\.\\d*", "FLOAT")
    lexer.add_rule("\\d+", "INT")
    lexer.add_rule("\\w+", "NAME")
    lexer.add_rule("\\+", "PLUS")
    lexer.add_rule("\\-", "MINUS")
    lexer.add_rule("\\*", "MUL")
    lexer.add_rule("/", "DIV")
    lexer.add_rule("=", "ASSIGN")
    lexer.add_rule("#[^\\r\\n]*", "COMMENT")
    lexer.add_ignores("[ \\f\\t]+")

    lexer.build(code)
    while True:
        token = lexer.next_token()
        if token.ttype != Token.TOKEN_TYPE_EOF:
            print(
                f"{token.name}\t{repr(token.source)}\tt{token.ttype}\t{token.position}\t{token.row},{token.col}")
        if token.ttype == Token.TOKEN_TYPE_EOF or token.ttype == Token.TOKEN_TYPE_ERROR:
            break


if __name__ == "__main__":
    main()
